File: LinkedList/SingleLinkedList.php
<?php

class Node{
    public mixed $data;
    public ?Node $next;

    public function __construct(mixed $data) {
        $this->data = $data;
        $this->next = null;
    }

}

class LinkedList {
    private ?Node $head;
    public int $length;

    public function __construct()
    {
        $this->head = null;
        $this->length = 0;
    }

    public function append(mixed $data, ?string $options = 'end'): void {
        $node = new Node($data);

        if ($options == 'front') {
            $node->next = $this->head;
            $this->head = $node;
            $this->length++;
            return;
        }

        if (!$this->head) {
            $this->head = $node;
            $this->length++;
            return;
        }

        $temp = $this->head;
        while($temp->next) {
            $temp = $temp->next;
        }
        $temp->next = $node;
        $this->length++;
        return;

    }

    public function reverse(): Node {
        $reverseLL = null;
        $temp = $this->head;
        while($temp) {
            $next_node = $temp->next;
            $temp->next = $reverseLL;
            $reverseLL = $temp;
            $temp = $next_node;
        }
        return $reverseLL;
    }

    public function print(): void {
        $node = $this->head;
        while($node) {
            echo $node->data .' ';
            $node = $node->next;
        }
    }

    public function printReverse(): void {
        $node = $this->reverse();
        while($node) {
            echo $node->data .' ';
            $node = $node->next;
        }
    }

    public function size(): int {
        return $this->length;
    }

}

function main(): void {

    $ll = new LinkedList();
    $lld = new LinkedList();
    $lla = new LinkedList();
    $llf = new LinkedList();

    $arr = [6,4,3,1,10,8,9,12,0];

    for($i = 0; $i <= 5; $i++) {
        $ll->append($i);
        $lld->append($i, options: 'front');
    }

    foreach($arr as $i) {
        $lla->append($i);
        $llf->append($i, options: 'front');
    }

    $ll->print();
    echo PHP_EOL;
    $ll->printReverse();
    echo PHP_EOL .$ll->length .PHP_EOL;
    echo $ll->size() .PHP_EOL;

    echo PHP_EOL;

    $lld->print();
    echo PHP_EOL;
    $lld->printReverse();
    echo PHP_EOL .$lld->length .PHP_EOL;
    echo $lld->size() .PHP_EOL;

    echo PHP_EOL;

    $lla->print();
    echo PHP_EOL;
    $lla->printReverse();
    echo PHP_EOL .$lla->length .PHP_EOL;
    echo $lla->size() .PHP_EOL;

    echo PHP_EOL;

    $llf->print();
    echo PHP_EOL;
    $llf->printReverse();
    echo PHP_EOL .$llf->length .PHP_EOL;
    echo $llf->size() .PHP_EOL;

}

main();

File: LinkedList/SingleLinkedList.py
from typing import Optional, TypeVar


T = TypeVar("T")


class Node:
    def __init__(self, data: T) -> None:
        self.data: T = data
        self.next: T = None

class SingleLinkedList:
    def __init__(self):
        self.head: Optional[Node] = None

    def append(self, data: T, position: str = "end") -> None:
        new_node: Node = Node(data)

        if position == "front":
            new_node.next = self.head
            self.head = new_node
            return

        if not self.head:
            self.head = new_node
            return

        temp = self.head
        while temp.next:
            temp = temp.next
        temp.next = new_node
        return

    def reverse(self) -> Node:
        new_list: Node = None
        temp = self.head
        while temp:
            next_node = temp.next
            temp.next = new_list
            new_list = temp
            temp = next_node
        return new_list

    def print_forward(self) -> None:
        node = self.head
        print('[', end='')
        while node:
            print(node.data, end=',')
            node = node.next
        print(']')

    def print_backward(self) -> None:
        node = self.reverse()
        print('[', end='')
        while node:
            print(node.data, end=',')
            node = node.next
        print(']')


if __name__ == '__main__':
    s = SingleLinkedList()
    sb = SingleLinkedList()

    for i in range(10):
        s.append(i)
    s.print_forward()
    s.print_backward()

    print()

    for i in range(10):
        sb.append(i, position="front")
    sb.print_forward()
    sb.print_backward()
